packages/fxa-settings/package.json

```diff
@@ -78,6 +78,7 @@
     "@types/react-webcam": "^3.0.0",
     "base32-decode": "^1.0.0",
     "base32-encode": "^1.2.0",
+    "class-validator": "^0.14.0",
```

Comment:
Following up on [this comment](https://github.com/mozilla/fxa/pull/15493#issuecomment-1631674385), any other pros/cons worth mentioning here? Anything you gleaned while working on this can help inform FXA-8100.

Reply:
I think there's a few points:

- The class validator gives us a lot of bang for the buck and paired well the the `@bind` decorator
- We can still validate on read/write access via some trickery in the model provider
- We can write custom validations if needed to reduce repetition. Personally I like the obvious nature of just using the standard validation set they provided.
- The error messaging and context for the validator is pretty nice

One downside:
- The entire model has to be validated. You can't target a specific field. There are 'validation groups' to validate a subset of a class's fields, but this is pretty noisy to apply at a granular level or a field by field basis.